content/blog/2024/jupyterhub-binderhub-gesis/index.md

```diff
@@ -0,0 +1,126 @@
+
+[cryptnono](https://github.com/cryptnono/cryptnono) is an open source project designed to help fight cryptojacking, and as part of this grant we ported some of this functionality out of mybinder.org specific code into cryptnono, so other deployments may also benefit from it! We also migrated to using the super efficient [ebpf](https://ebpf.io/) Linux Kernel subsystem, allowing for more complex heuristics to catch a much broader range of cryptomining activity. We have been slowly tweaking the config on mybinder.org, and it has proven to be very effective! This will be very helpful for *anyone* who wants to provide a JupyterHub (or any other computational service) without a login wall. If you are interested in using cryptnono in this fashion, please [reach out to us](https://github.com/cryptnono/cryptnono/issues) so we can work together!
+
+## Explored pathways that were then discarded
+
+List of things that were tried and then decided as not good pathways:
+
+- [repo2docker-service](https://github.com/consideRatio/repo2docker-service), a separate JupyterHub service that could *only* build images. As we worked on it, we realized that it was replicating a lot of features that BinderHub already has, so we pivoted to working on BinderHub directly instead.
+- Building off of [tljh-repo2docker](https://github.com/plasmabio/tljh-repo2docker). While this already had a nice UI, it would be hard to port it to run on a distributed kubernetes environment without it becoming a 'hard fork'.
+
+While these did slow down the implementation of the project, it has allowed us to be very confident that the methods we have chosen are long term sustainable.
+
+## Want to try this out?
+
+We have a demo of this running at [imagebuilding-demo.2i2c.cloud](https://imagebuilding-demo.2i2c.cloud), but unfortunately as we are still fine-tuning `cryptnono` config, at this moment it is not open to the public. Please [hit me up](mailto:yuvipanda@2i2c.org) with your GitHub account if you want access, and promise to not be a cryptominer and you shall be granted access.
+
+Want to set this up on your own JupyterHub? There is some [work in progress](https://github.com/2i2c-org/2i2c-org.github.io/pull/205) documentation and more is being worked on. Drop a line in the linked pull request and we'll be happy to help. The eventual goal is for *anyone* to be able to simply follow documentation and set this up for themselves.
+
+## Future work
```

Comment:
Is the authorization aspect between components resolved already? Otherwise that can be listed here also perhaps.

Reply:
Added!